src/scanner-utils.h

```diff
@@ -218,6 +236,23 @@ scanner_hex(struct scanner *s, uint8_t *out)
     return i > 0;
 }
 
+/** Parser for \u{NNNN} escape sequences */
+static inline bool
+scanner_unicode(struct scanner *s, uint32_t *out)
+{
+    uint8_t i;
+    uint32_t u = 0;
+    for (i = 0; is_xdigit(scanner_peek(s)) && i < 9; i++) {
```

Comment:
Out of curiosity, I checked and:

- Python has `\uXXXX` and `\UXXXXXXXX` (exact amount of digits)
- Rust has `\u{XXXXXXX}` (up to 6 digits)

This looks like the Rust syntax (which I agree is somewhat nicer than Python one), but up to 9 digits. Maybe better to restrict to 6 to keep compat with Rust?

Reply:
Done. The previous version was indeed a mix of the Rust and C99 syntax.